hack/snapshot_to_catalog.sh

```diff
@@ -0,0 +1,138 @@
+echo "using yq from ${YQ}"
+# check if yq exists
+if [ -z "${YQ}" ]; then
+  echo "yq is required"
+  exit 1
+fi
+
+BUNDLE_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-operator-bundle"
+OPERATOR_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-rhel9-operator"
+CONSOLE_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-console-plugin-rhel9"
+SERVICE_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-service-api-rhel9"
+
+BUNDLE_IMAGE_ORIGIN=$(jq -r '.spec.components[]| select(.name=="bundle") | .containerImage' ${TMP_SNAPSHOT_JSON})
+BUNDLE_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/bundle|'"${BUNDLE_IMAGE_BASE}"'|g' <<<${BUNDLE_IMAGE_ORIGIN})
+
+OPERATOR_IMAGE=$(jq -r '.spec.components[]| select(.name=="lightspeed-operator") | .containerImage' ${TMP_SNAPSHOT_JSON})
+OPERATOR_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-operator|'"${OPERATOR_IMAGE_BASE}"'|g' <<<${OPERATOR_IMAGE})
+
+CONSOLE_IMAGE=$(jq -r '.spec.components[]| select(.name=="lightspeed-console") | .containerImage' ${TMP_SNAPSHOT_JSON})
+CONSOLE_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-console|'"${CONSOLE_IMAGE_BASE}"'|g' <<<${CONSOLE_IMAGE})
+
+SERVICE_IMAGE=$(jq -r '.spec.components[]| select(.name=="lightspeed-service") | .containerImage' ${TMP_SNAPSHOT_JSON})
+SERVICE_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-service|'"${SERVICE_IMAGE_BASE}"'|g' <<<${SERVICE_IMAGE})
+
+RELATED_IMAGES=$(
+  cat <<-EOF
+[
+  {
+    "name": "lightspeed-service-api",
+    "image": "${SERVICE_IMAGE}"
+  },
+  {
+    "name": "lightspeed-console-plugin",
+    "image": "${CONSOLE_IMAGE}"
+  },
+  {
+    "name": "lightspeed-operator",
+    "image": "${OPERATOR_IMAGE}"
+  },
+  { "name": "lightspeed-operator-bundle",
+    "image": "${BUNDLE_IMAGE}"
+  }
+]
+EOF
+)
+
+echo "Catalog will use the following images:"
+echo "BUNDLE_IMAGE=${BUNDLE_IMAGE}"
+echo "OPERATOR_IMAGE=${OPERATOR_IMAGE}"
+echo "CONSOLE_IMAGE=${CONSOLE_IMAGE}"
+echo "SERVICE_IMAGE=${SERVICE_IMAGE}"
+
+echo BUNDLE_IMAGE_ORIGIN=${BUNDLE_IMAGE_ORIGIN}
+${OPM} render ${BUNDLE_IMAGE_ORIGIN} --output=yaml >"${TEMP_BUNDLE_FILE}"
+
+# restore bundle image to the catalog file
+${YQ} eval -i '.image='"\"${BUNDLE_IMAGE}\"" ${TEMP_BUNDLE_FILE}
+# restore bundle related images and the bundle itself to the catalog file
+${YQ} eval -i '.relatedImages='"${RELATED_IMAGES}" ${TEMP_BUNDLE_FILE}
+
+#Initialize lightspeed-catalog/index.yaml from hack/operator.yaml
+CATALOG_INITIAL_FILE="hack/operator.yaml"
+cat ${CATALOG_INITIAL_FILE} >${CATALOG_FILE}
+
+cat ${TEMP_BUNDLE_FILE} >>${CATALOG_FILE}
+
+cat <<EOF >>"${CATALOG_FILE}"
+---
+schema: olm.channel
+package: lightspeed-operator
+name: preview
+entries:
+  - name: lightspeed-operator.v${BUNDLE_TAG}
+EOF
+
+${OPM} validate $(dirname "${CATALOG_FILE}")
```

Comment:
I'd quote $(dirname) to avoid splitting

Reply:
What kind of splitting dirname can give us?

Reply:
Do you mean dirname can return "(" and ")"?

Reply:
added a quote in the new commit, is that what you mean?